### PR TITLE
Add missing description for discount types.

### PR DESCRIPTION
**Pull request by DevilsAutumn**
I want to merge this change because it adds missing descriptions to discount types/

Refs #13142 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/discount/types/vouchers.py [hunk 1]
```diff
@@ -51,17 +61,31 @@ def resolve_channel(root: models.VoucherChannelListing, info: ResolveInfo):
...
-    apply_once_per_order = graphene.Boolean(required=True)
-    apply_once_per_customer = graphene.Boolean(required=True)
-    only_for_staff = graphene.Boolean(required=True)
-    min_checkout_items_quantity = graphene.Int()
+    id = graphene.GlobalID(required=True, description="The ID of the voucher.")
+    name = graphene.String(description="The name of the voucher.")
+    code = graphene.String(required=True, description="The code of the voucher.")
+    usage_limit = graphene.Int(description="The number of times a voucher can be used.")
+    used = graphene.Int(
+        required=True, description="Determine if the voucher has been used."
```
Comment (korycins) [inline saleor/graphql/discount/types/vouchers.py:69]: The field stores the number of voucher usage, so you can determine if the voucher has been used but also you can see how many times it was used.
I would change the description to something like:
`"The number of times a voucher has been used."`

File: saleor/graphql/discount/types/vouchers.py [hunk 2]
```diff
@@ -51,17 +61,31 @@ def resolve_channel(root: models.VoucherChannelListing, info: ResolveInfo):
...
+    )
+    start_date = graphene.DateTime(
+        required=True, description="The start date and time of voucher."
+    )
+    end_date = graphene.DateTime(description="The end date and time of voucher.")
+    apply_once_per_order = graphene.Boolean(
+        required=True,
+        description="Determine if the voucher should be applied once per order.",
```
Comment (korycins) [inline saleor/graphql/discount/types/vouchers.py:77]: I think that we can improve the description here. 
Something like `Determine if the voucher should be applied only to a single cheapest eligible product`

File: saleor/graphql/discount/types/vouchers.py [hunk 3]
```diff
@@ -51,17 +61,31 @@ def resolve_channel(root: models.VoucherChannelListing, info: ResolveInfo):
...
+    )
+    apply_once_per_customer = graphene.Boolean(
+        required=True,
+        description="Determine if the voucher should be applied once per order.",
```
Comment (korycins) [inline saleor/graphql/discount/types/vouchers.py:81]: Incorrect (looks like copied from once_per_order). 
It should be something like:
"Determine if the voucher usage should be limited to one use per customer"

Comment (DevilsAutumn) [reply saleor/graphql/discount/types/vouchers.py:81]: agh, my bad.

File: CHANGELOG.md [hunk 4]
```diff
@@ -84,7 +84,8 @@ All notable, unreleased changes to this project will be documented in this file.
 - Add `ACCOUNT_CONFIRMATION_REQUESTED` async event - #13162 by @SzymJ
 - Add `ACCOUNT_DELETE_REQUESTED` async event - #13170 by @SzymJ
 - Add `ACCOUNT_CHANGE_EMAIL_REQUESTED` async event - #13233 by @SzymJ
-
+- Add missing descriptions to Discount module - #13261 by @devilsautumn
+-
```
Comment (maarcingebala) [inline CHANGELOG.md:88]:
```suggestion

```

File: saleor/graphql/discount/types/sales.py [hunk 5]
```diff
@@ -35,10 +35,20 @@
 
 
 class SaleChannelListing(ModelObjectType[models.SaleChannelListing]):
-    id = graphene.GlobalID(required=True)
-    channel = graphene.Field(Channel, required=True)
-    discount_value = graphene.Float(required=True)
-    currency = graphene.String(required=True)
+    id = graphene.GlobalID(required=True, description="The ID of the channel list.")
```
Comment (maarcingebala) [inline saleor/graphql/discount/types/sales.py:38]:
```suggestion
    id = graphene.GlobalID(required=True, description="The ID of the channel listing.")
```

File: saleor/graphql/discount/types/sales.py [hunk 6]
```diff
@@ -51,13 +61,19 @@ def resolve_channel(root: models.SaleChannelListing, info: ResolveInfo):
...
-    id = graphene.GlobalID(required=True)
-    name = graphene.String(required=True)
-    type = SaleType(required=True)
-    start_date = graphene.DateTime(required=True)
-    end_date = graphene.DateTime()
-    created = graphene.DateTime(required=True)
-    updated_at = graphene.DateTime(required=True)
+    id = graphene.GlobalID(required=True, description="The ID of the sale.")
+    name = graphene.String(required=True, description="The name of the sale.")
+    type = SaleType(required=True, description="Fixed or percent")
```
Comment (maarcingebala) [inline saleor/graphql/discount/types/sales.py:66]:
```suggestion
    type = SaleType(required=True, description="Type of the sale, fixed or percentage.")
```

File: saleor/graphql/discount/types/vouchers.py [hunk 7]
```diff
@@ -34,11 +34,21 @@
 
 
 class VoucherChannelListing(ModelObjectType[models.VoucherChannelListing]):
-    id = graphene.GlobalID(required=True)
-    channel = graphene.Field(Channel, required=True)
-    discount_value = graphene.Float(required=True)
-    currency = graphene.String(required=True)
-    min_spent = graphene.Field(Money)
+    id = graphene.GlobalID(required=True, description="The ID of channel list.")
```
Comment (maarcingebala) [inline saleor/graphql/discount/types/vouchers.py:37]:
```suggestion
    id = graphene.GlobalID(required=True, description="The ID of channel listing.")
```

File: saleor/graphql/discount/types/vouchers.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/discount/types/vouchers.py:69]: Usage count of the voucher.

File: saleor/graphql/discount/types/vouchers.py (hunk 2 above)
Comment (maarcingebala) [reply saleor/graphql/discount/types/vouchers.py:77]: Determine if the voucher should be applied once per order. If set to True, the voucher is applied to a single cheapest eligible product in checkout.

File: saleor/graphql/discount/types/vouchers.py [hunk 8]
```diff
@@ -51,17 +61,31 @@ def resolve_channel(root: models.VoucherChannelListing, info: ResolveInfo):
...
+    apply_once_per_order = graphene.Boolean(
+        required=True,
+        description="Determine if the voucher should be applied once per order.",
+    )
+    apply_once_per_customer = graphene.Boolean(
+        required=True,
+        description="Determine if the voucher should be applied once per order.",
+    )
+    only_for_staff = graphene.Boolean(
+        required=True, description="Determine if the voucher is only for staff."
```
Comment (maarcingebala) [inline saleor/graphql/discount/types/vouchers.py:84]: Determine if the voucher is available only for staff members.

**Review by maarcingebala — APPROVED** [review]

**Review by korycins — APPROVED** [review]
Good work! 🙌  Thanks!

**Comment by maarcingebala** [comment]
@DevilsAutumn Please re-generate the GraphQL schema as it's outdated now.

**Comment by DevilsAutumn** [comment]
@maarcingebala Done.

**Comment by maarcingebala** [comment]
@DevilsAutumn Thanks! I recommend installing [pre-commit](https://docs.saleor.io/docs/3.x/developer/community/contributing#coding-style) which runs the same checks as the CI, but locally before each commit. Helps to find such issues much earlier.

**Comment by DevilsAutumn** [comment]
> @DevilsAutumn Thanks! I recommend installing [pre-commit](https://docs.saleor.io/docs/3.x/developer/community/contributing#coding-style) which runs the same checks as the CI, but locally before each commit. Helps to find such issues much earlier.

I did install pre-commit, but graphql-schema-check was being stuck everytime(maybe because schema is too large?). So i had to use `git commit no-verify`

**Pull request merged**